### PR TITLE
[harness-simulation] update `deviceInputFields.xml` automatically

### PR DESCRIPTION
**Pull request by jcdong98**
This commit automatically updates the device fields in `deviceInputFields.xml`, eliminating the need to update them manually.

### CODE REVIEW

File: tools/harness-simulation/harness/Web/data/updateDeviceFields.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def main():
+    tree = ET.parse(HARNESS_XML_PATH)
+    root = tree.getroot()
+    added = ET.parse(sys.argv[1]).getroot()
+
+    added_names = set(device.attrib['name'] for device in added.iter('DEVICE'))
+    # If some devices already exist, remove them first, and then add them back in case of update
+    removed_devices = [device for device in root.iter('DEVICE') if device.attrib['name'] in added_names]
```
Comment (superwhd) [inline tools/harness-simulation/harness/Web/data/updateDeviceFields.py:15]: nit: **might** be more concise to use `filter`

File: tools/harness-simulation/harness/Web/data/updateDeviceFields.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+HARNESS_XML_PATH = r'%s\GRL\Thread1.2\Web\data\deviceInputFields.xml' % os.environ['systemdrive']
```
Comment (superwhd) [inline tools/harness-simulation/harness/Web/data/updateDeviceFields.py:5]: Maybe we should also make this a command line argument. For example, do we need to switch to `Thread 1.3` in the future?

Comment (simonlingoogle) [reply tools/harness-simulation/harness/Web/data/updateDeviceFields.py:5]: The path is fixed. It's always Thread1.2 for all 1.1/1.2/1.3 certifications.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

after adding license to tools/harness-simulation/harness/Web/data/updateDeviceFields.py

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**